src/Microsoft.DotNet.ProjectModel/Files/PatternGroup.cs

```diff
@@ -12,6 +12,8 @@ namespace Microsoft.DotNet.ProjectModel.Files
 {
     public class PatternGroup
     {
+        private static readonly Dictionary<string, IEnumerable<string>> s_resolvedFilesCache = new Dictionary<string, IEnumerable<string>>();
```

Comment:
Definitely this area is a bit concerning.  If you cache the `SearchFiles` results statically, and a file is added/removed from the file system in between calls, you aren't going to be picking up the new changes to the file system.

I think you'll need to refactor the code to keep the same `PatternGroup` instance around for the life of the build/publish/etc command.  I don't think this is the right design to cache the results statically.